grpc.go

```diff
@@ -502,21 +502,25 @@ func (a *agentGRPC) WaitProcess(ctx context.Context, req *pb.WaitProcessRequest)
 		return &pb.WaitProcessResponse{}, err
 	}
 
-	defer func() {
-		proc.closePostExitFDs()
-		ctr.deleteProcess(proc.id)
-	}()
-
-	// Using helper function wait() to deal with the subreaper.
-	libContProcess := (*reaperLibcontainerProcess)(&(proc.process))
-	exitCode, err := a.sandbox.subreaper.wait(proc.exitCodeCh, libContProcess)
-	if err != nil {
-		return &pb.WaitProcessResponse{}, err
+	select {
+	case exitCode := <-proc.exitCodeCh:
+		// proc.process.Wait() & proc.closePostExitFDs()
+		// should be called only once.
```

Comment:
Using https://golang.org/pkg/sync/#Once might be faster for subsequent calls, but the locking here ensures safety.